### PR TITLE
Update: architecture.md for merge server

### PR DESCRIPTION
**Pull request by Jiaaming**

### CODE REVIEW

**Comment by Jiaaming** [comment]
Screenshot 2024-07-22 at 16 24 53

File: ARCHITECTURE.md [hunk 1]
```diff
@@ -7,17 +7,28 @@ This extension contains three major components:
...
-<img src="images/gradle-tasks-architecture.svg" />
+<img src="images/gradle-server-architecture.svg" />
+The gradle server is a long-running Java process that include two threads: 1. Build Server 2. Task Server
 
-The extension uses client/server architecture using [gRPC](https://grpc.io/) as the interface between the client and server. It uses TypeScript (Node.js) on the client and Java on the server. A long running server provides very good performance.
+## Build Server
+The Gradle Build Server uses the [Build Server Protocol](https://build-server-protocol.github.io/) to communicate with the server.
+
+Historically, the Build Client and Build Server were connected via standard input/output, which posed security concerns. To address these issues and accommodate limitations with named pipe support in Java and JSON-RPC connections (based on the Java implementation [bsp4j](https://github.com/build-server-protocol/build-server-protocol/) of the Build Server Protocol, which itself relies on [lsp4j](https://github.com/eclipse-lsp4j/lsp4j)), a TypeScript layer named [bspProxy](./extension/src/bs/bspProxy.ts) was introduced.
+
```
Comment (Jiaaming) [inline ARCHITECTURE.md]: bspProxy -> BspProxy. Will fix

File: ARCHITECTURE.md [hunk 2]
```diff
@@ -7,17 +7,28 @@ This extension contains three major components:
 
 # Gradle Server and Gradle Language Server
 
-<img src="images/gradle-tasks-architecture.svg" />
+<img src="images/gradle-server-architecture.svg" />
+The gradle server is a long-running Java process that include two threads: 1. Build Server 2. Task Server
 
-The extension uses client/server architecture using [gRPC](https://grpc.io/) as the interface between the client and server. It uses TypeScript (Node.js) on the client and Java on the server. A long running server provides very good performance.
+## Build Server
+The Gradle Build Server uses the [Build Server Protocol](https://build-server-protocol.github.io/) to communicate with the server.
```
Comment (jdneo) [inline ARCHITECTURE.md]: with the server?

**Review by jdneo — APPROVED** [review]
LGTM

**Pull request merged**